datatable/src/main/java/io/cucumber/datatable/DataTableFormatter.java

```diff
@@ -65,23 +65,31 @@ public void formatTo(DataTable table, Appendable appendable) throws IOException
             }
         }
 
+        // Buffer the output with a StringBuilder, to avoid excess calls to
+        // flush() from the NiceAppender used by PrettyFormatter.
+        // Gradle adds a newline for every call to flush() unless the most
+        // recent character written was already a newline. That leads
+        // to some VERY difficult to read tables on the console.
+        StringBuilder buffer = new StringBuilder();
```

Comment:
Normally it could be assumed that the `Appendable` is well behaved.  However for the formatter we know that lines may interleave with other output as it writes to `System.out`. Consider pushing this buffer up to the pretty formatter instead.

Reply:
Pushed the buffer up to the pretty formatter by commit f4a2d42 . Only caveat - I can't decide how to test that particular change or whether a unit test is warranted.